shared/login/signup/request-invite.render.desktop.js

```diff
@@ -0,0 +1,90 @@
+/* @flow */
+/* eslint-disable react/prop-types */
+
+import React, {Component} from 'react'
+import {globalStyles} from '../../styles/style-guide'
+import {Text, Icon, Input, Button} from '../../common-adapters'
+import Container from '../forms/container'
+
+import type {Props} from './request-invite.render'
+
+export default class Render extends Component {
+  props: Props;
+
+  state: {
+    email: ?string,
+    name: ?string
+  };
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {
+      email: this.props.email || '',
+      name: this.props.name || ''
+    }
+  }
+
+  render () {
+    const submitRequestInvite = () => {
+      this.props.onRequestInvite(this.state.email, this.state.name)
+    }
+
+    return (
+      <Container onBack={this.props.onBack} style={styles.container}>
+        <Text style={styles.header} type='Header'>Request an invite code</Text>
+        <Icon style={styles.icon} type='invite-code-m'/>
+        <Input
+          style={styles.input}
+          floatingLabelText='Your email address'
+          value={this.state.email || ''}
+          errorText={this.props.emailErrorText}
+          onChange={event => this.setState({email: event.target.value})}
+        />
+        <Input
+          style={styles.input}
+          floatingLabelText='Your name'
+          value={this.state.name || ''}
```

Comment:
That's what you specified ;) 
https://github.com/keybase/client/blame/master/shared/common-adapters/input.js.flow#L18

You defined this type as an optional key, but if it is there it should be a string.

If you don't want that you can say I want an optional key and if it is there it is either a string or null.
That would look like: `value?: ?string`.